corehq/apps/userreports/tests/test_expressions.py

```diff
@@ -1073,7 +1073,7 @@ def test_invalid_eval_expression(self, source_doc, statement, context):
     ("a+b+c+9", {"a": 5, "b": 2, "c": 8}, 5 + 2 + 8 + 9),
     ("a*b", {"a": 2, "b": 23}, 2 * 23),
     ("a*b if a > b else b -a", {"a": 2, "b": 23}, 23 - 2),
-    ("'text1' if a < 5 else `text2`", {"a": 4}, 'text1'),
```

Comment:
doesn't this test something else now? should it look like ```"'text1' if a < 5 else \`text2\`"```?

Reply:
I think the only reason this ever worked was due to short-circuit evaluation in the py2 version of ```simpleeval```:

Py2:
```
>>> from simpleeval import SimpleEval
>>> SimpleEval().eval("'abc' if True else `text`")
'abc'
>>> SimpleEval().eval("'abc' if False else `text`")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/Nick/.virtualenvs/feb_2018/lib/python2.7/site-packages/simpleeval.py", line 274, in eval
    return self._eval(ast.parse(expr.strip()).body[0].value)
  File "/Users/Nick/.virtualenvs/feb_2018/lib/python2.7/site-packages/simpleeval.py", line 285, in _eval
    return handler(node)
  File "/Users/Nick/.virtualenvs/feb_2018/lib/python2.7/site-packages/simpleeval.py", line 337, in _eval_ifexp
    else self._eval(node.orelse)
  File "/Users/Nick/.virtualenvs/feb_2018/lib/python2.7/site-packages/simpleeval.py", line 283, in _eval
    "evaluator".format(type(node).__name__))
simpleeval.FeatureNotAvailable: Sorry, Repr is not available in this evaluator
>>> 
```

Py3:
```
>>> from simpleeval import SimpleEval
>>> SimpleEval().eval("'abc' if True else `text`")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/Nick/.virtualenvs/py3-2/lib/python3.6/site-packages/simpleeval.py", line 274, in eval
    return self._eval(ast.parse(expr.strip()).body[0].value)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/ast.py", line 35, in parse
    return compile(source, filename, mode, PyCF_ONLY_AST)
  File "<unknown>", line 1
    'abc' if True else `text`
                       ^
SyntaxError: invalid syntax
>>> SimpleEval().eval("'abc' if False else `text`")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/Nick/.virtualenvs/py3-2/lib/python3.6/site-packages/simpleeval.py", line 274, in eval
    return self._eval(ast.parse(expr.strip()).body[0].value)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/ast.py", line 35, in parse
    return compile(source, filename, mode, PyCF_ONLY_AST)
  File "<unknown>", line 1
    'abc' if False else `text`
                        ^
SyntaxError: invalid syntax
```

@sravfeyn I saw you added this test 3 years ago https://github.com/dimagi/commcare-hq/commit/cdba3d60802567465aea668557c2ffb1ed7795e4 - do you think this change is ok?  Unless the original intention was to test short-circuit evaluation with broken syntax, I think this change is ok.

Reply:
Oh I see now. I think `text` is valid python2 syntax and it would get the `__repr__` of the variable:

```
>>> blah = 2
>>> `blah`
'2'
```

It looks like that was removed in python 3. and it likely wouldn't have worked anyways because you can't specify variables in UCR anyways